src/app/container/container.component.ts

```diff
@@ -197,8 +197,16 @@ export class ContainerComponent extends Entry implements AfterViewInit {
     if (tool) {
       this.tool = tool;
       this.initTool();
+      this.requestAccessHREF$ = this.extendedTool$.pipe(
+        map((data) => {
+          if (data.imgProvider) {
+            return this.emailService.composeRequestAccessEmail(this.tool, data.imgProvider);
+          } else {
+            return this.emailService.composeRequestAccessEmail(this.tool, 'undefined');
```

Comment:
this entire section should just be one return statement. the logic for checking imgProvider should be done in getRequestEmailBody